File: src/main/java/com/battletrident/listeners/EventManager.java
package com.battletrident.listeners;

import com.battletrident.listeners.event.*;
import com.battletrident.listeners.event.gui.OnAdminGUI;
import com.battletrident.listeners.event.skill.*;
import org.bukkit.Bukkit;
import org.bukkit.event.Listener;

import static com.battletrident.consts.Plugin.Plugin;

public class EventManager {
	private final Listener[] LISTENERS = {
		new OnAdminGUI(),

		new OnGameUpdate(),
		new OnRingUpdate(),
		new OnPlayerInteract(),
		new OnPlayerDeath(),

		new OnTrident(),
		new OnCinderella(),
		new OnBangal(),
	};

	public EventManager() {
		for (Listener listener : LISTENERS) {
			Bukkit.getPluginManager()
				.registerEvents(listener, Plugin);
		}
	}
}

File: src/main/kotlin/com/battletrident/listeners/EventManager.kt
package com.battletrident.listeners

import com.battletrident.BattleTrident.Companion.plugin
import com.battletrident.guis.adminGUI.settingGUI.SettingGUIListener
import com.battletrident.listeners.event.*
import com.battletrident.listeners.event.gui.AdminGUIListener
import com.battletrident.skills.cinderella.gui.CinderellaGUIListener
import com.battletrident.skills.bangal.OnBangal
import com.battletrident.skills.cinderella.OnCinderella
import com.battletrident.skills.trident.OnTrident

class EventManager {
	val LISTENERS = listOf(
		OnPlayerJoinExit(),
		OnGameUpdate(),
		OnRingUpdate(),
		OnPlayerInteract(),
		OnPlayerDeath(),
		
		AdminGUIListener(),
		SettingGUIListener(),

		OnTrident(),
		OnCinderella(),
		OnBangal(),
		
		CinderellaGUIListener(),
	)

	init {
		val pm = plugin.server.pluginManager
		
		for (listener in LISTENERS) {
			pm.registerEvents(listener, plugin)
		}
	}
}